util/network-devp2p/src/node_record.rs

```diff
@@ -9,26 +10,37 @@ const ENR_VERSION: &str = "v4";
 pub struct EnrManager {
 	secret: secp256k1::SecretKey,
 	inner: Enr,
+	path: Option<PathBuf>,
 }
 
 impl EnrManager {
-    pub fn new(key: Secret, seq: u64) -> Option<Self> {
+	fn save(&mut self) {
+		if let Some(path) = &self.path {
```

Comment:
It's a bit worrying that this fails silently.

Reply:
Freestanding `save` issues a warning on failure.